File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)

        postfix = [-1]*n
        cur_max = arr[-1]
        for i in range(n-2,-1,-1):
            postfix[i]=cur_max
            cur_max = max(arr[i],cur_max)
            
        
        return postfix

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        return binarySearch(nums,target,0,nums.length-1);
    }
    public int binarySearch(int[] arr,int target,int start,int end){
        if(end-start<1)
            return target==arr[end] ? end : -1;
        int mid=start+((end-start)/2);
        return Math.max(binarySearch(arr,target,start,mid),binarySearch(arr,target,mid+1,end));
    }
}
